Sources/ProtobufferCoding/ProtoCodingKey.swift

```diff
@@ -0,0 +1,15 @@
+//
+//  ProtoCodable.swift
+//  
+//
+//  Created by Moritz Schüll on 19.11.20.
+//
+
+import Foundation
+
+
+protocol ProtoCodingKey: CodingKey {
+    static func mapCodingKey(_ key: CodingKey) throws -> Int?
```

Comment:
I am unsure about the ergonomics and naming of this static method here.

Alternatives:
```
extension ProtoCodingKey {
    var protoRawValue: Result<Int?, Error> {
        return .success(0)
    }
    
    func protoRawValue() throws -> Int? {
        0
    }
}
```

Furthermore, do we really want to allow Optional\<Int\>? Most of the time, coding throws an error, if it is not present, i.e., in `KeyedProtoDecodingContainer.swift`.
I am just currently not aware of the differences between `return nil` or `throw Error`- 🙂

Reply:
These are two valid points 👍
Your suggested name `protoRawValue` is definitely more telling. I also agree that we should not use both optional and throws; will get rid of the optionals.